pkg/payload/precondition/clusterversion/upgradeable.go

```diff
@@ -42,16 +43,43 @@ func (pf *Upgradeable) Run(ctx context.Context) error {
 			Name:    pf.Name(),
 		}
 	}
-	if up := resourcemerge.FindOperatorStatusCondition(cv.Status.Conditions, configv1.OperatorUpgradeable); up != nil && up.Status == configv1.ConditionFalse {
-		return &precondition.Error{
-			Nested:  err,
-			Reason:  up.Reason,
-			Message: up.Message,
-			Name:    pf.Name(),
-		}
+
+	// if we are upgradeable==true we can always upgrade
+	up := resourcemerge.FindOperatorStatusCondition(cv.Status.Conditions, configv1.OperatorUpgradeable)
+	if up == nil || up.Status != configv1.ConditionFalse {
+		return nil
+	}
+
+	// we can always allow the upgrade if there isn't a version already installed
+	if len(cv.Status.History) == 0 {
+		return nil
+	}
+
+	currentMinor := getEffectiveMinor(cv.Status.History[0].Version)
```

Comment:
Instead of pulling this out of the cluster's ClusterVersion history, I'd rather pull it from [the local state][1].  But then you'd have to pipe it through to this precondition code.  Including it in `ReleaseContext` would be fine, although we'd want to rename to something more generic.  Maybe just `Context` (so `precondition.Context` in external packages)?  To get from the `Operator` into the `SyncWorker`... I dunno, seems like we could feed the context into `NewSyncWorkerWithPreconditions` [here][2]?  If that sounds plausible, I can work up a fixup commit that you could squash in here.

[1]: https://github.com/openshift/cluster-version-operator/blob/9281b45f9ab65a92b1dd4fe263683a6d60628d48/pkg/cvo/cvo.go#L628-L630
[2]: https://github.com/openshift/cluster-version-operator/blob/9281b45f9ab65a92b1dd4fe263683a6d60628d48/pkg/cvo/cvo.go#L274

Reply:
It's not the way I would do it since we're already getting the information here (pre-existing gets) and you're already getting cached data, so it seems unnecessary and odd.  Could you do it as a follow-up instead if you really want it after?